iroha/src/queue.rs

```diff
@@ -1,17 +1,32 @@
+use self::config::QueueConfiguration;
 use crate::prelude::*;
 
-#[derive(Default, Debug)]
+#[derive(Debug)]
 pub struct Queue {
     pending_tx: Vec<AcceptedTransaction>,
+    max_block_tx: usize,
 }
 
 impl Queue {
+    pub fn from_configuration(config: &QueueConfiguration) -> Queue {
+        Queue {
+            pending_tx: Vec::new(),
+            max_block_tx: config.max_block_tx as usize,
+        }
+    }
+
     pub fn push_pending_transaction(&mut self, tx: AcceptedTransaction) {
         self.pending_tx.push(tx);
     }
 
     pub fn pop_pending_transactions(&mut self) -> Vec<AcceptedTransaction> {
-        self.pending_tx.drain(..).collect()
+        let len = self.pending_tx.len();
+        let n = if self.max_block_tx > len {
+            len
+        } else {
+            self.max_block_tx
+        };
+        self.pending_tx.drain(..n).collect()
```

Comment:
@eadventurous gave variables good names - pending_transactions_length, amount_to_drain, and maximum_transactions_in_block